manifests/plugin/discovery.pp

```diff
@@ -33,15 +33,19 @@
   if $install_images {
     include ::tftp::params
 
-    foreman::remote_file {"${::tftp::params::root}boot/${kernel}":
-      remote_location => "${source}${kernel}",
+    foreman::remote_file {"${::tftp::params::root}boot/${image_name}":
```

Comment:
Btw, since https://github.com/theforeman/puppet-tftp/commit/c0bc540d542ea2f2e037aad4926b32dc64d685f1 it can be overridden and before that you could already refer to ```$tftp::root``` due to inheritance so let's use that instead.